ecosystem/typescript/sdk/src/aptos_client.ts

```diff
@@ -914,6 +914,19 @@ export class AptosClient {
     return this.client.blocks.getBlockByVersion(version, withTransactions);
   }
 
+  /**
+   * Call for a move view function
+   *
+   * @param payload Transaction payload
+   * @param version (optional) Ledger version to lookup block information for
+   *
+   * @returns MoveValue[]
+   */
+  @parseApiError
+  async view(payload: Gen.ViewRequest, ledger_version?: string): Promise<Gen.MoveValue[]> {
```

Comment:
noob question: How is this ViewRequest generated?

Reply:
I believe anything in `Gen` comes from codegen based on the OpenAPI spec from the API.

Reply:
using OpenAPI
https://github.com/aptos-labs/aptos-core/tree/main/ecosystem/typescript/sdk#generating-api-client
specifically
https://github.com/aptos-labs/aptos-core/blob/main/api/doc/spec.yaml#L8513